packages/xo-server/src/xapi/mixins/storage.mjs

```diff
@@ -49,18 +49,19 @@ export default {
     await this._unplugPbd(this.getObject(id))
   },
 
-  _getVdiChainsInfo(uuid, childrenMap, cache) {
+  _getVdiChainsInfo(uuid, childrenMap, cache, resultContainer) {
     let info = cache[uuid]
     if (info === undefined) {
       const children = childrenMap[uuid]
       const unhealthyLength = children !== undefined && children.length === 1 ? 1 : 0
+      resultContainer.nUnhealthyVdis += unhealthyLength
```

Comment:
Just to make sure I understood the change: you're counting on the fact that when we encounter the same VDI again, this SR-level counter won't be incremented because we'll hit the cache line 53, correct?